### DISCUSSION TITLE
Kyuubi on K8S with Spark SQL engine - spark.kubernetes.executorEnv are not getting set or pass onto Spark Executors

### DISCUSSION THREAD

**Discussion by mmuru** [apache/kyuubi#6465] [Q&A] [answered]
Using latest Kyuubi release 1.9.1 + Spark-3.5.1 + Kubernetes 1.25

I tried to setup environment variables in the both Spark Driver and executor pods by setting spark.kubernetes.driverEnv and spark.kubernetes.executorEnv in the spark-default.conf. After the spark driver and executors are running, noticed only the environment variables set in the drive pod but not in the executor pods. In the driver, I do see them in the spark.properties file in the /opt/spark/conf and the same file does not exists in the /opt/spark/conf on the executors. 

Is it a known issues with Kyuubi + Spark on kubernetes? Otherwise, please let me know how to set environments in the executors.

Thanks in advance

**Comment by pan3793** [comment, marked as answer]
There is no `spark.kubernetes.executorEnv.[EnvironmentVariableName]` configuration, only `spark.executorEnv.[EnvironmentVariableName]`

https://spark.apache.org/docs/latest/configuration.html

**Reply by mmuru** [reply]
@pan3793 : You're right. I didn't  pay attention the Spark documentation. Somehow I missed the below notes. 

`Spark will add driver env variables from spark.kubernetes.driverEnv.[EnvironmentVariableName], and executor env variables from spark.executorEnv.[EnvironmentVariableName].`

After setting spark.executorEnv.<var> and it worked as expected.

Thanks for your help.
